source/renderer/app/components/staking/delegation-setup-wizard/DelegationStepsChooseWalletDialog.js

```diff
@@ -126,39 +142,46 @@ export default class DelegationStepsChooseWalletDialog extends Component<
       getStakePoolById,
     } = this.props;
 
-    const selectedWallet: ?Wallet =
-      wallets.find(
-        (wallet: Wallet) => wallet && wallet.id === selectedWalletId
-      ) || null;
-    const amount = selectedWallet ? selectedWallet.amount : null;
-    const isAcceptableSetupWallet = amount && isWalletAcceptable(amount);
+    const selectedWallet: ?Wallet = wallets.find(
+      (wallet: Wallet) => wallet && wallet.id === selectedWalletId
+    );
+
+    const { amount, reward, isRestoring } = selectedWallet || {};
+
+    let errorMessage;
+    if (selectedWallet && !isWalletAcceptable(amount, reward)) {
+      // Wallet is restoring
+      if (isRestoring) errorMessage = messages.errorRestoringWallet;
+      // Wallet only has Reward balance
+      else if (!amount.isZero() && amount.equals(reward))
+        errorMessage = messages.errorMinDelegationFundsRewardsOnly;
+      // Wallet balance - rewards < min delegation funds
+      /*
+      else if (!reward.isZero())
+        errorMessage = messages.errorMinDelegationFundsHasRewards;
+      */
```

Comment:
@daniloprates What is this? Do we need to handle `Wallet balance - rewards < min delegation funds` or you just left a comment for some future improvement?

Reply:
Yes, I was waiting on Darko's position on this. He's just given thumbs up, so I'll get copy/translation and uncomment it on a separate card.

Reply:
@daniloprates Amazing, thanks.
You did an AMAZING  job on this PR 💯 🚀